Problems with Linux

There's an error when compiling under Linux (I'm using Ubuntu 20.04). Line 827 in `src/engine/ArduinoSerial.cpp` references a macro that is only defined for OSX - `if (ioctl(portDescriptor, IOSSIOSPEED, &speed) == -1)`. I got things compiling and running by commenting this section and uncommenting the section above ("the comment described it as the traditional setup of baud rates").

That led to another error - I find that when I run SpikeRecorder, it starts up just fine, but it doesn't detect my MuscleBox plugged in. If I try with the Windows version of SpikeRecorder in a virtual machine, it works ok.

Strange... BYBBootloaderController.h is included under the directory.
src/engine/firmware/BYBBootloaderController.h  maybe try adding:
-Isrc/engine/firmware

On Tue, May 31, 2022 at 8:15 AM Gemarcm ***@***.***> wrote:

> I can't compile Spikerecorder in my pop-os 22.04. The followint error
> after typing make.
>
> make
> g++ -o src/engine/AudioInputConfig.o -c src/engine/AudioInputConfig.cpp -g
> -O2 -Isrc -Isupport -Isrc/libraries -I. -Lsrc/support -Wall
> -DSIGSLOT_PURE_ISO --std=c++11 sdl2-config --cflags
> g++ -o src/CalibrationWindow.o -c src/CalibrationWindow.cpp -g -O2 -Isrc
> -Isupport -Isrc/libraries -I. -Lsrc/support -Wall -DSIGSLOT_PURE_ISO
> --std=c++11 sdl2-config --cflags
> In file included from src/CalibrationWindow.cpp:6:
> src/engine/RecordingManager.h:19:10: fatal error:
> BYBBootloaderController.h: No such file or directory
> 19 | #include "BYBBootloaderController.h"
> | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
> compilation terminated.
> make: *** [Makefile:78: src/CalibrationWindow.o] Error 1
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/BackyardBrains/Spike-Recorder/issues/44#issuecomment-1142055776>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAG52575UEGHZL3YTACPAQTVMX7GRANCNFSM5NGPEVBA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Gregory Gage, Ph.D.
Backyard Brains, Inc.
***@***.***
mob: +1 (734) 968-7570
DIY Neuroscience *TED* Series: https://www.ted.com/series/diy_neuroscience
Human *TED* talk: https://www.youtube.com/watch?v=rSQNi5sAwuc
Plant *TED* talk: http://go.ted.com/greggage
Cockroach *TED* talk: https://www.ted.com/talks/the_cockroach_beatbox
Neuroscience for Everyone!

@gagegreg Thanks for the help, I add the path as you recommended, from directory: firmware/BYBBootloaderController.h only.
Then I got a similar error and I had to add the same path (firmware/BYBBootloaderController.h) in the ArduinoSerial.cpp and it fixed it. It kept compiling until it gave me this new error and I have no idea how to fix it. 

gema@pop-os:~/reposgit/Spike-Recorder$ make
g++ -o src/engine/ArduinoSerial.o -c src/engine/ArduinoSerial.cpp -g -O2 -Isrc -Isupport -Isrc/libraries -I. -Lsrc/support -Wall -DSIGSLOT_PURE_ISO --std=c++11 `sdl2-config --cflags`
src/engine/ArduinoSerial.cpp:165: warning: ignoring ‘#pragma mark ’ [-Wunknown-pragmas]
  165 | #pragma mark - Port scanning and opening
      | 
src/engine/ArduinoSerial.cpp:1622: warning: ignoring ‘#pragma mark ’ [-Wunknown-pragmas]
 1622 | #pragma mark - Reading and processing
      | 
src/engine/ArduinoSerial.cpp:2598: warning: ignoring ‘#pragma mark ’ [-Wunknown-pragmas]
 2598 | #pragma mark - Getters/parameters
      | 
src/engine/ArduinoSerial.cpp:2639: warning: ignoring ‘#pragma mark ’ [-Wunknown-pragmas]
 2639 | #pragma mark - Messages - Write
      | 
src/engine/ArduinoSerial.cpp: In member function ‘int BackyardBrains::ArduinoSerial::openPort(const char*)’:
src/engine/ArduinoSerial.cpp:886:36: error: ‘IOSSIOSPEED’ was not declared in this scope
  886 |          if (ioctl(portDescriptor, IOSSIOSPEED, &speed) == -1) {
      |                                    ^~~~~~~~~~~
src/engine/ArduinoSerial.cpp: In member function ‘int BackyardBrains::ArduinoSerial::processDataIntoSamples(char*, int, int16_t*)’:
src/engine/ArduinoSerial.cpp:1812:13: warning: unused variable ‘g’ [-Wunused-variable]
 1812 |         int g=0;
      |             ^
src/engine/ArduinoSerial.cpp:1813:14: warning: unused variable ‘enablePrint’ [-Wunused-variable]
 1813 |         bool enablePrint = false;
      |              ^~~~~~~~~~~
make: *** [Makefile:78: src/engine/ArduinoSerial.o] Error 1

Hello again, I have done everything you said, and it keep compiling until this new error. I'll really appreciate your help here. 

/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/11/../../../../lib/libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /lib/../lib/libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /usr/lib/../lib/libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/11/../../../libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /lib/libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /usr/lib/libbass.so when searching for -lbass
/usr/bin/ld: cannot find -lbass: No such file or directory
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/11/../../../../lib/libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /lib/../lib/libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /usr/lib/../lib/libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/11/../../../libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /lib/libbass.so when searching for -lbass
/usr/bin/ld: skipping incompatible /usr/lib/libbass.so when searching for -lbass
collect2: error: ld returned 1 exit status
make: *** [Makefile:81: SpikeRecorder] Error 1

--- a/Makefile
+++ b/Makefile
@@ -16,6 +16,9 @@ OBJECTS = \
 		src/engine/FileRecorder.o \
 		src/engine/Player.o \
 		src/engine/ArduinoSerial.o \
+		src/engine/firmware/BYBBootloaderController.o \
+		src/engine/WavTxtRecorder.o \
+		src/FirmwareUpdateView.o \
 		src/engine/SpikeSorter.o \
 		src/engine/SpikeAnalysis.o \
 		src/engine/BASSErrors.o \
@@ -68,7 +71,7 @@ OBJECTS_LINUX = \
 
 OBJECTS += $(OBJECTS_LINUX)
 
-CFLAGS = -g -O2 -Isrc -Isupport -Isrc/libraries -I. -Lsrc/support -Wall -DSIGSLOT_PURE_ISO --std=c++11 `sdl2-config --cflags`
+CFLAGS = -g -O2 -Isrc -Isupport -Isrc/libraries -I. -Isrc/engine/firmware -Lsrc/support -Wall -DSIGSLOT_PURE_ISO --std=c++11 `sdl2-config --cflags`
 LIBS = `sdl2-config --libs` -lSDL2_image -lGL -lGLU -lbass -lpthread -lhidapi-libusb
 
 %.o: %.c

--- a/src/engine/ArduinoSerial.cpp
+++ b/src/engine/ArduinoSerial.cpp
@@ -31,6 +31,7 @@
     #include <linux/serial.h>
     #include <cstring>
     #include <cstdio>
+    #define IOSSIOSPEED _IOW('T', 2, speed_t)
 
 #elif defined(__APPLE__)
 //added for port scan

--- a/src/engine/firmware/BYBBootloaderController.cpp
+++ b/src/engine/firmware/BYBBootloaderController.cpp
@@ -119,7 +119,7 @@ namespace BackyardBrains {
             parseLineOfHex(contents);
             //free(contents);
 
-        }
+        
 
 
         initTransferOfFirmware();
@@ -233,10 +233,7 @@ namespace BackyardBrains {
         #endif // defined
         #if defined(__linux__)
             int bits;
-            if (size == 0 && ioctl(fd, TIOCMGET, &bits) < 0)
-            {
-                //error
-            }
+            
         #endif
         #if defined(_WIN32)
             COMSTAT st;

